src/proto/h1/dispatch.rs

```diff
@@ -180,12 +186,29 @@ where
         task::yield_now(cx).map(|never| match never {})
     }
 
+    fn check_header_read_timeout(&mut self, cx: &mut task::Context<'_>) -> crate::Result<()> {
+        if self.header_read_timeout.is_some() {
+            if let Some(read_timeout_fut) = &mut self.header_read_timeout_fut {
+                if Pin::new(read_timeout_fut).poll(cx).is_ready() {
+                    warn!("read header from client timeout");
+                    return Err(crate::Error::new_header_timeout());
+                }
+            } else {
+                self.header_read_timeout_fut = Some(Box::pin(sleep(self.header_read_timeout.unwrap())))
+            }
+        }
+        Ok(())
```

Comment:
You can remove the use of unwrap by using `if let`

```suggestion
        if let Some(header_read_timeout) = self.header_read_timeout {
            if let Some(read_timeout_fut) = &mut self.header_read_timeout_fut {
                if Pin::new(read_timeout_fut).poll(cx).is_ready() {
                    warn!("read header from client timeout");
                    return Err(crate::Error::new_header_timeout());
                }
            } else {
                self.header_read_timeout_fut = Some(Box::pin(sleep(header_read_timeout)))
            }
        }

        Ok(())
```